src/action/payment.js

```diff
@@ -174,17 +174,26 @@ class PaymentAction {
       this._nav.goWait();
       const invoice = this._store.payment.address.replace(PREFIX_URI, '');
       const stream = this._grpc.sendStreamCommand('sendPayment');
-      await new Promise((resolve, reject) => {
+      const success = await new Promise((resolve, reject) => {
+        const timeout = setTimeout(() => resolve(false), PAYMENT_TIMEOUT);
         stream.on('data', data => {
           if (data.payment_error) {
+            clearTimeout(timeout);
```

Comment:
Looks like we're clearing the timeout here regardless of the conditional. Can we move it outside the conditional (below line 179) and delete line 184?